server/apps/item_lock/components/item_lock.py

```diff
@@ -80,7 +80,8 @@ def unlock(self, item_filter, user_id, session_id, etag):
             self.app.on_item_unlock(item, user_id)
 
             # delete the item if nothing is saved so far
-            if item[config.VERSION] == 0 and item['state'] == 'draft':
+            # version 1 created on lock item
+            if item[config.VERSION] == 1 and item['state'] == 'draft':
```

Comment:
It should be version 0 only. For items created in personal desk it starts with version 0. 

@petrjasek correct me if am wrong.